crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -169,7 +188,19 @@ impl MsgArea {
     }
 
     pub(crate) fn page_down(&mut self) {
-        self.scroll = max(0, self.scroll - 10);
+        self.scroll.scroll = max(0, self.scroll.scroll - 10);
+    }
+
+    /// Determines if we need to recalculate the scroll offset due to resizing of the window
+    fn recalculate_scroll(&mut self) -> bool {
+        if self.scroll.scroll > 0 {
+            // Subtract current height since our self.scroll.line_idx is the top of the window,
+            // but self.scroll.scroll is the offset from the bottom
```

Comment:
I'm curious, why is this inconsistency exists? Why not make both hold the bottom/offset from bottom or top/offset from top?

Reply:
I couldn't sort out how to do the scroll recalculation correctly while keeping the top-most line at the top of the screen during a resize (and also scrolled). You mentioned that it will be good to have it stick there and reveal newer messages when the screen is expanded.

This was easy to use since `self.scroll.line_idx` is set during `draw()` (not nice to have a side effect), where the visible lines are drawn in reverse, from bottom to top. We end on the last visible index (`line_idx`) of `self.lines` and set `self.scroll.line_idx` to that.